Blog/starter pack (#5553)
sourcegraph/about: content/blogposts/blog-post-starter-pack.md

        <td>5.1</td>
        <td>3.5</td>
        <td>1.4</td>
        <td>0.2</td>
      </tr>
      <tr>
        <td>2</td>
        <td>Iris Setosa</td>
        <td>4.9</td>
        <td>3.0</td>
        <td>1.4</td>
        <td>0.2</td>
      </tr>
    </tbody>
  </table>
</TableWrapper>

### YouTube

You can easily embed YouTube videos using the `YouTube` component. The `id` value is found at the end of the URL for any YouTube video. To display the title below the video, add the `showTitle` property.

<YouTube
  title="Accelerate developer onboarding with Sourcegraph" 
  id="DgwvhRW1Cbc"
  showTitle={true}
/>

### Video

The `Video` component supports .mp4 and .webm files. First, upload the two versions of the video to Google Cloud. Then, add the root names of these files to the `source` property, without including their extensions. You can choose to repeat a video by including the `loop` property, or display the caption using the `showCaption` property.

<Video 
  source={{webm: 'blog/shift-left/01-code-reuse', mp4: 'blog/shift-left/01-code-reuse'}} 
  loop={true}
  title="Code reuse"
  caption="An example of code search with Sourcegraph"
  showCaption={true}
/>

> ### HubSpot Form Example
> 
> Use the `EmbeddedHubSpot` component to drop a custom HubSpot form into your post. In most cases, the `formId` is the only property that needs to be updated. This is the same `formId` dictated by the HubSpot console. Once added to your post, the HubSpot form will render wherever you place the `div` using the specified `id` value.
> 
> <EmbeddedHubSpot 
>   portalId='2762526' 
>   formId='08e6c442-0e7c-4892-a262-76dae55ab497' 
>   targetId='#sign-up-form' />
> 
> <div id="sign-up-form"></div>

Reviewer: This API has changed and the component is now called `HubSpotForm`

```suggestion
### HubSpot Form Example

Use the `HubSpotForm` component to drop a custom HubSpot form into your post. In most cases, the `masterFormName` and `chiliPiper` props are the only properties that need to be updated. Once added to your post, the HubSpot form will render in its place. All the options for this component are:

- `formId` - an optional form ID if the form doesn't fall under one of our [master forms](https://docs.google.com/document/d/1qa393uAW_tjwNtH9jHf_d__PCJgmbsxNoxi9cO4XOcE/edit)
- `masterFormName` - an optional master form name; `contactMulti`, `contactEmail`, `gatedMulti`, or `gatedEmail` for general contact forms and gated forms that have multi fields or an email field
- `onFormSubmitted` - an optional callback function for triggering an action after a form is submitted
- `inlineMessage` - an optional message to display after a form is submitted. We have a [generic fallback](https://github.com/sourcegraph/about/blob/main/src/components/HubSpotForm.tsx#L288) if none is provided. 
- `chiliPiper` - an optional boolean (`true`/`false`) to enable Chili Piper. This is false by default.

<HubSpotForm
  masterFormName="contactEmail"
  chiliPiper={true}
/>
```

Reviewer 2: Thank you!

## Code

> Any code you write, you can style! The Blog uses Prism for code formatting. Specify the language at the front of the code block. 

Author: Totally agree with you here. This is much better!

> Any code you write, you can style! The Blog uses Prism for code formatting. Specify the language at the front of the code block. 

Reviewer 2: @bretthayes , is there a way to make sure this mock blog post won't publish to prod?

**TypeScript**

```ts
export const BlogPost<BlogPostProps> = ({title, text}) = {
  <h1>{title}</h1>
  <p>{text}</p>
  …
}
```

## Links

Add links with an anchor tag as you usually would: 

<div className="text-center mb-5">
  <a href="https://www.sourcegraph.com/search">Check out Sourcegraph Search!</a>
</div>

If you are wrapping an image in a link, please include a `title` on the anchor tag for accessibility.

> <a href="https://www.sourcegraph.com/search" title="Go to Sourcegraph Code Search">

Reviewer 2: We should try to make the titles also describe the image along with the link since the screen reader will skip over the image when the code is written this way. It may take some trial and error to see what the screen reader says & what makes sense.

Author: Great point! What do you think of adding this context to the description, then altering the link title and image alt text? Maybe something like this:

```suggestion
If you are wrapping an image in a link, please include a `title` on the anchor tag for accessibility, this way screen readers can pick up all the content. Screen readers may skip over the image, so it is important for the `title` to include a description of the image and intended action.

<a href="https://www.sourcegraph.com/search" title="An image of the Sourcegraph logo that links to Sourcegraph Code Search">
  <Figure 
    src="https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png"
    alt="Sourcegraph logo thumbnail"
  />
</a>
```

Reviewer 2: So the screen reader will read the link text & the image alt text only if the anchor tag surrounds a pure img tag, but when it surrounds the our Figure component it doesn't. Have you tested it? You can test with this blog `/blog/optimizing-a-code-intel-backend`. I think we should reconfigure the `Figure` component to make it more accessible, or encourage blog writers to not make complex images a link. Or instruct if the image is also a link, just to use the pure `<img/ >` tag instead of Figure. What do you think?

Reviewer: I agree! I don't think we need to link Figures or images even @st0nebraker but maybe include a link the caption/cite if need be.

Reviewer 2: Update made [8889b16](https://github.com/sourcegraph/about/pull/5553/commits/8889b16002511333223064eaba8e642c711bf206) :)

description: This is a template for writing content for the Sourcegraph blog!
authors:
  - name: Sourcegraph
    url: https://twitter.com/sourcegraph
publishDate: 2022-07-01T18:00+02:00
tags: [blog]
slug: starter-pack
published: true
---
```

### `title` & `description`

The `title` will automatically appear at the top of your post. The description will appear on the published list of posts at `/blog`.

### `slug`

Determine the URL for your post using the `slug`. For example, `slug: my-blog-post` will set the URL to `/blog/my-blog-post`.

### `author`

Blog posts support multiple authors. Add another author using `- name`.

### `published`

A post with `published: true` will appear in the list of published posts at `/blog`. To unpublish a post, set `published: false`.

### `socialImage` & `heroImage`

To set the thumbnails for this post, link to an image file using `socialImage` and `heroImage`. By default, blog posts use the Sourcegraph OG image. Using these properties in the frontmatter overrides the default. 

- `socialImage` determines the image appearing on shareable links
- `heroImage` determines the cover image appearing on `/blog`

<hr/>

## Components

> The Sourcegraph Blog is implemented in Next.js and MDX. One strength of this implementation is the ability to incorporate components in Markdown. The following components are available for blog posts:

Reviewer: ```suggestion
The Sourcegraph Blog is implemented with Next.js with MDX support. One strength of this implementation is the ability to incorporate our custom components in Markdown. The following components are available for blog posts:
```

1. Alert
2. Blockquote
3. Figure
4. TableWrapper
5. Video

> 6. YouTube

Reviewer: ```suggestion
6. YouTube, and;
```

7. HubSpotForm

> Use these components to bring our blog post to life. Here are a few examples:

Reviewer: ```suggestion
Use these components to bring your blog post to life. Here are a few examples:
```

This table is written in **Markdown**:

<TableWrapper>
| Id   | Species     | Sepal Length (cm) | Sepal Width (cm) | Petal Length (cm) | Petal Width (cm) |
|------|-------------|-------------------|------------------|-------------------|------------------|
| 1    | Iris Setosa | 5.1               | 3.5              | 1.4               | 0.2              |
| 2    | Iris Setosa | 4.9               | 3.0              | 1.4               | 0.2              |
</TableWrapper>

This is the same table, but written in **HTML**:

<TableWrapper>
  <table>
    <thead>
      <tr>
        <th>ID</th>
        <th>Name</th>
        <th>Sepal Length (cm)</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>1</td>
        <td>Iris Setosa</td>
        <td>5.1</td>
        <td>3.5</td>
        <td>1.4</td>
        <td>0.2</td>
      </tr>
      <tr>
        <td>2</td>
        <td>Iris Setosa</td>
        <td>4.9</td>
        <td>3.0</td>
        <td>1.4</td>
        <td>0.2</td>
      </tr>
    </tbody>
  </table>
</TableWrapper>

### YouTube

> You can easily embed YouTube videos using the `YouTube` component. The `id` value is found at the end of the URL for any YouTube video. To display the title below the video, add the `showTitle` property.

Reviewer: ```suggestion
You can easily embed YouTube videos using the `YouTube` component. The two required properties are `title` and `id`.

A title should be descriptive for your YouTube video and is used for accessibility and the caption. It is also hidden by default. To optionally display the title below the video as a caption, add the `showTitle` property.

The `id` value is found at the end of the URL for any YouTube video.
```

<YouTube
  title="Accelerate developer onboarding with Sourcegraph" 
  id="DgwvhRW1Cbc"
  showTitle={true}
/>

### Video

> The `Video` component supports .mp4 and .webm files. First, upload the two versions of the video to Google Cloud. Then, add the root names of these files to the `source` property, without including their extensions. You can choose to repeat a video by including the `loop` property, or display the caption using the `showCaption` property.

Reviewer: The API will change slightly here in #5615

```suggestion
The required properties for the `Video` component are `mp4`, `webm`, `loop`, `caption`, and `title`. 

Since the component supports both `.mp4` and `.webm` files, you'll need to upload the two versions of the video to our Google Cloud [assets bucket](https://console.cloud.google.com/storage/browser/sourcegraph-assets/blog) under an appropriate sub directory. Then, add the file path from the Public URL of these files to the `mp4` and `webm` properties without their file extensions.

You can choose to repeat a video by including the `loop` property, optionally auto play the video by including the `autoPlay` property, or optionally display the caption using the `showCaption` property.
```

> <Video 
>   source={{webm: 'blog/shift-left/01-code-reuse', mp4: 'blog/shift-left/01-code-reuse'}} 
>   loop={true}
>   title="Code reuse"
>   caption="An example of code search with Sourcegraph"
>   showCaption={true}
> />

Reviewer: ```suggestion
<Video 
  mp4="/blog/shift-left/01-code-reuse"
  webm="/blog/shift-left/01-code-reuse"
  loop={true}
  title="Code reuse"
  caption="An example of code search with Sourcegraph"
  showCaption={true}
/>
```

Reviewer 2: Since this isn't in prod yet it's a breaking change right now. I'll bring it in once it's shipped!

- `masterFormName` - an optional master form name; `contactMulti`, `contactEmail`, `gatedMulti`, or `gatedEmail` for general contact forms and gated forms that have multi fields or an email field
- `onFormSubmitted` - an optional callback function for triggering an action after a form is submitted
- `inlineMessage` - an optional message to display after a form is submitted. We have a [generic fallback](https://github.com/sourcegraph/about/blob/main/src/components/HubSpotForm.tsx#L288) if none is provided. 
- `chiliPiper` - an optional boolean (`true`/`false`) to enable Chili Piper. This is false by default.

<HubSpotForm
  masterFormName="contactEmail"
  chiliPiper={true}
/>

## Code

You can add syntax highlighting to any code you write! The Blog uses Prism for code formatting. Specify the language at the front of the code block. 

**TypeScript**

```ts
export const BlogPost<BlogPostProps> = ({title, text}) = {
  <h1>{title}</h1>
  <p>{text}</p>
  …
}
```

## Links

Add links with an anchor tag as you usually would: 

<div className="text-center mb-5">
  <a href="https://www.sourcegraph.com/search">Check out Sourcegraph Search!</a>
</div>

> Instead of linking images, please link the image's caption. This will allow our Figure components to be more accessible to screen readers. To add a link to an image, provide a `caption` and `link` property to the `Figure` component.

Reviewer: ```suggestion
Instead of linking images using Markdown, please use the Figure component. This will allow our images to be more accessible to screen readers. To add a link to an image, provide a `caption` and `link` property to the `Figure` component.
```

<Figure 
  src="https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png"
  alt="Sourcegraph thumbnail"
  caption="Check out Sourcegraph Search!"
  link="https://www.sourcegraph.com/search"
/>

> ## About the author
> 
> _Author blurb information goes here. You can chat with [author] on Twitter [@author-name](https://twitter.com/) or our community [Discord](https://discord.com/invite/vqsBW8m5Y8)._

Reviewer: ```suggestion
## About the author

We encourage our authors to include an "about the author" section to their blog posts. Use the following Markdown for consistency throughout our blog posts.

_Author blurb information goes here. You can chat with [author] on Twitter [@author-name](https://twitter.com/) or our community [Discord](https://discord.com/invite/vqsBW8m5Y8)._
```

## TrySourcegraph CTA

> Note: A footer that includes the "Try Sourcegraph" CTA is automatically included at the end of each blog post.

Reviewer: ```suggestion
Note: A footer that includes the "Try Sourcegraph" CTA is automatically included at the end of each blog post so there is no need to include it manually.
```

<hr/>

## Need support?

> Please reach out to the Content Platform Team in #content-platform with any questions!

Reviewer: ```suggestion
If you have any questions, concerns, or requests, please reach out to the Content Platform Team in #content-platform!
```

> ---

Reviewer 2: Removing all these empty "Blogy McBlogerson" blogposts, no one knew about these blogposts. They probably outdate most of marketing and don't need to be in prod.

---
title: Blog Post Starter Pack
description: This is a template for writing content on the Sourcegraph blog!
authors:
  - name: Sourcegraph
    url: https://twitter.com/sourcegraph

> publishDate: 2013-01-01T18:00+02:00

Reviewer: Ref: [official founding date](https://docs.google.com/document/d/19lePEDt9S6GODakIfzoB1HDf9p8ls1etwuFpXzAMLSo/edit)

```suggestion
publishDate: 2013-01-19T00:00
```

tags: [blog]
slug: starter-pack
heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png
socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png
published: false
---

<Figure
  src="https://storage.googleapis.com/sourcegraph-assets/blog/default_hero_social.png"
  alt="Blog Post Starter Pack"
/>

<Alert>To preview this template visit [`/blog/starter-pack`](https://about.sourcegraph.com/blog/starter-pack).</Alert>

This template surfaces components and patterns commonly used in the Sourcegraph Blog. All of the sections below are ready for you to copy and paste into your blog post today!

To start a new blog post, create a new `[slug].md` file under the current year's sub directory within the `/content/blogposts/` directory.

<hr className="my-6" />

## Frontmatter

This section is called `frontmatter`, it provides the post's metadata.

```yaml
---
title: Blog Post Starter Pack
description: This is a template for writing content for the Sourcegraph blog!
authors:
  - name: The author name
    url: https://example.com/
  - name: Second authors name (optional)
    url: https://example-2.com/

> publishDate: 2022-07-01T18:00+02:00

Reviewer: Just for kicks 🥚🙊

```suggestion
publishDate: 2013-01-19T00:00
```

The `id` value is found at the end of the URL for any YouTube video.

<YouTube
  title="Accelerate developer onboarding with Sourcegraph" 
  id="DgwvhRW1Cbc"
  showTitle={true}
/>

### Video

The required properties for the `Video` component are `mp4`, `webm`, `loop`, `caption`, and `title`. 

Since the component supports both `.mp4` and `.webm` files, you'll need to upload the two versions of the video to our Google Cloud [assets bucket](https://console.cloud.google.com/storage/browser/sourcegraph-assets/blog) under an appropriate sub directory. Then, add the file path from the Public URL of these files to the `mp4` and `webm` properties without their file extensions.

You can choose to repeat a video by including the `loop` property, optionally auto play the video by including the `autoPlay` property, or optionally display the caption using the `showCaption` property.

<Video 
  source={{webm: 'blog/shift-left/01-code-reuse', mp4: 'blog/shift-left/01-code-reuse'}} 
  loop={true}
  title="Code reuse"
  caption="An example of code search with Sourcegraph"
  showCaption={true}

> />

Reviewer 2: This will cause the build to break though. When #5615 is merged I'll commit this. Or if this goes in first, #5615 can make this change to the file.

> />

Reviewer: @st0nebraker Yes I know, that's what I'm saying - we should wait for #5615 to go in first since we don't want to encourage people to write this code since it will break.

> />

Reviewer 2: Can you break the bug fix into a separate PR to separate concerns? @bretthayes

> />

Reviewer: @st0nebraker I updated the language slightly in the suggestion and reverted back to the original syntax since I confirmed that it's working in #5619
